weave

```diff
@@ -806,6 +806,19 @@ dns_args() {
     fi
 }
 
+rewrite_hosts() {
+    hosts_path_and_name=$(docker inspect -f '{{.HostsPath}} {{.Config.Hostname}}' $CONTAINER)
+    hosts=$(echo $hosts_path_and_name | cut -f1 -d" ")
+    name=$(echo $hosts_path_and_name | cut -f2 -d" ")
+    containers_path="$(dirname $(dirname "$hosts"))"
```

Comment:
Well, it does the same thing [launch_proxy](https://github.com/weaveworks/weave/blob/master/weave#L1203) does... Only one `dirname` would result in `/containers/[hosts, resolv.conf, ...]`, so `/containers` wouldn't contain _containers_ anymore but only the current container...  I don't mind if you prefer it this way...